src/main/scala/com/target/data_validator/EnvironmentVariables.scala

```diff
@@ -0,0 +1,58 @@
+package com.target.data_validator
+
+import scala.collection.mutable
+import scala.util.Try
+
+object EnvironmentVariables {
+  type MaybeEnvVar = Try[Option[String]]
+
+  val accessedEnvVars: mutable.Map[String, MaybeEnvVar] = mutable.Map.empty
+  private[this] def clearAccessList(): Unit = accessedEnvVars.clear()
+
+  def get(key: String): EnvVarResult = {
+    getWithHandlers(key)(
+      whenError = { case throwable: Throwable => Inaccessible(throwable) },
+      whenUnset = { Unset },
+      whenPresent = { Present }
+    )
+      .recover { case throwable: Throwable => Error(throwable) }
+      .get
+  }
+
+  def getWithHandlers[T](key: String)(whenError: PartialFunction[Throwable, T],
+                                      whenUnset: => T,
+                                      whenPresent: String => T
+  ): Try[T] = {
```

Comment:
Can you indent this consistent with the rest of the code base?  Put first arg on a new line and use 2 space indenting.